### DISCUSSION TITLE
FullPath option in mod.json

### DISCUSSION THREAD

**Discussion by IceReaper** [voxeltycoon/issues#453] [Modding]
I would suggest to add another property to ```mod.json``` namely ```"FullPath": true``` which defaults to false.
Having this option makes the assets in the mod use their full path (relative to the mod dir, using the standard ```/``` seperator which works under windows too) instead of their filename only pattern.
This can be extremely usefull if a mod reaches a high complexity state and ships dozens of assets or if a particular mod has buildin support of swappable assets by swapping the load folder.

**Comment by victorleblais** [comment]
I totally agree with your suggestion.

I'm also thinking of an implementation to simplify references by using a specific keyword/syntax.

For example, inside my "refreshfr_decorative_ballast" mod, I currently have:
`"MeshUri": "refreshfr_decorative_ballast/2x1_diagonal_ballast_patch.obj",`
instead, it could be simplified to
`"MeshUri": "@/2x1_diagonal_ballast_patch.obj",`

If I'm referencing an asset from a different file, I could write
`"MeshUri": "@rubber_tires_system/rubber_tires_regular_1.obj",`

Basically, with a simple regex, you can check the string between the `@` and the first `/`, if it's empty, then it's referencing itself, if not, then it's referencing another module (and since a mod name can't be empty, it's not an issue to have `@/`, whereas a specific keywork like `_self/` or something else could be problematic)

I don't know, it's just an idea ! :P
